### ISSUE TITLE
mamba cannot find providers for python >=3.7,<3.8.0a0 when installing rsat-core

### ISSUE THREAD

**Issue by apposada** [mamba-org/mamba#937]
Hi,

On a fresh installation of mamba in a test environment, Mamba says that it was not able to find python >=3.7,<3.8.0a0 .

```
$ mamba install -c bioconda -c conda-forge rsat-core

                  __    __    __    __
                 /  \  /  \  /  \  /  \
                /    \/    \/    \/    \
███████████████/  /██/  /██/  /██/  /████████████████████████
              /  / \   / \   / \   / \  \____
             /  /   \_/   \_/   \_/   \    o \__,
            / _/                       \_____/  `
            |/
        ███╗   ███╗ █████╗ ███╗   ███╗██████╗  █████╗
        ████╗ ████║██╔══██╗████╗ ████║██╔══██╗██╔══██╗
        ██╔████╔██║███████║██╔████╔██║██████╔╝███████║
        ██║╚██╔╝██║██╔══██║██║╚██╔╝██║██╔══██╗██╔══██║
        ██║ ╚═╝ ██║██║  ██║██║ ╚═╝ ██║██████╔╝██║  ██║
        ╚═╝     ╚═╝╚═╝  ╚═╝╚═╝     ╚═╝╚═════╝ ╚═╝  ╚═╝

        mamba (0.12.2) supported by @QuantStack

        GitHub:  https://github.com/mamba-org/mamba
        Twitter: https://twitter.com/QuantStack

█████████████████████████████████████████████████████████████


Looking for: ['rsat-core']

bioconda/linux-64        [====================] (00m:01s) Done
bioconda/noarch          [====================] (00m:01s) Done
pkgs/main/linux-64       [====================] (00m:01s) Done
pkgs/r/linux-64          [====================] (00m:00s) Done
pkgs/r/noarch            [====================] (00m:00s) Done
pkgs/main/noarch         [====================] (00m:00s) Done
conda-forge/noarch       [====================] (00m:02s) Done
conda-forge/linux-64     [====================] (00m:04s) Done
Encountered problems while solving:
  - package rsat-core-2020.01.04-py37pl526r36hc99cbb1_0 requires python >=3.7,<3.8.0a0, but none of the providers can be installed
```

Despite this, it seems to be found when running repoquery:

```
$ mamba repoquery depends -c bioconda -c conda-forge rsat-core

                  __    __    __    __
                 /  \  /  \  /  \  /  \
                /    \/    \/    \/    \
███████████████/  /██/  /██/  /██/  /████████████████████████
              /  / \   / \   / \   / \  \____
             /  /   \_/   \_/   \_/   \    o \__,
            / _/                       \_____/  `
            |/
        ███╗   ███╗ █████╗ ███╗   ███╗██████╗  █████╗
        ████╗ ████║██╔══██╗████╗ ████║██╔══██╗██╔══██╗
        ██╔████╔██║███████║██╔████╔██║██████╔╝███████║
        ██║╚██╔╝██║██╔══██║██║╚██╔╝██║██╔══██╗██╔══██║
        ██║ ╚═╝ ██║██║  ██║██║ ╚═╝ ██║██████╔╝██║  ██║
        ╚═╝     ╚═╝╚═╝  ╚═╝╚═╝     ╚═╝╚═════╝ ╚═╝  ╚═╝

        mamba (0.12.2) supported by @QuantStack

        GitHub:  https://github.com/mamba-org/mamba
        Twitter: https://twitter.com/QuantStack

█████████████████████████████████████████████████████████████


Executing the query rsat-core

bioconda/linux-64        Using cache
bioconda/noarch          Using cache
conda-forge/linux-64     Using cache
conda-forge/noarch       Using cache


 Name                          Version    Build                       Channel             
───────────────────────────────────────────────────────────────────────────────────────────
 rsat-core                     2020.02.29 py36pl526r36hc99cbb1_0      bioconda/linux-64   
 numpy                         1.10.4     py27_blas_openblas_200      conda-forge/linux-64
 scipy                         0.17.1     np110py27_blas_openblas_200 conda-forge/linux-64
 matplotlib                    1.5.3      np110py27_0                 conda-forge/linux-64
 r-ggplot2                     2.0.0      r3.2.2_0                    bioconda/linux-64   
 pyyaml                        3.11       py27_0                      conda-forge/linux-64
 r-rcolorbrewer                1.1_2      r3.3.2_0                    conda-forge/linux-64
 r-data.table                  1.10.0     r3.3.1_0                    bioconda/linux-64   
 bedtools                      2.16.2     0                           bioconda/linux-64   
 r-reshape2                    1.4.2      r3.3.2_0                    conda-forge/linux-64
 r-dplyr                       0.5.0      r3.3.1_0                    bioconda/linux-64   
 blast                         2.10.1     pl526he19e7b1_0             bioconda/linux-64   
 r-gplots                      2.17.0     r3.3.1_0                    bioconda/linux-64   
 r-gridextra                   2.2.1      r3.3.2_0                    conda-forge/linux-64
 snakemake                     3.10.0     py34_0                      bioconda/linux-64   
 perl-bioperl                  1.6.924    1                           bioconda/linux-64   
 perl-json                     2.90       0                           bioconda/linux-64   
 perl-dbi                      1.634      1                           bioconda/linux-64   
 perl-yaml                     1.15       0                           bioconda/linux-64   
 ghostscript                   9.07       0                           bioconda/linux-64   
 perl-cgi                      4.22       1                           bioconda/linux-64   
 perl-xml-parser               2.44       1                           bioconda/linux-64   
 gnuplot                       4.6.0      1                           bioconda/linux-64   
 r-devtools                    1.12.0     r3.3.2_0                    conda-forge/linux-64
 perl-db-file                  1.835      4                           bioconda/linux-64   
 bioconductor-qvalue           2.10.0     r3.4.1_0                    bioconda/linux-64   
 perl-parallel-forkmanager     1.17       0                           bioconda/linux-64   
 perl-log-log4perl             1.47       0                           bioconda/linux-64   
 r-png                         0.1.7      r3.2.2_0                    bioconda/linux-64   
 perl-file-spec                3.48_01    pl5.22.0_0                  bioconda/linux-64   
 perl-data-dumper              2.161      pl5.22.0_0                  bioconda/linux-64   
 perl-dbd-mysql                4.033      1                           bioconda/linux-64   
 perl-lwp-simple               6.15       1                           bioconda/linux-64   
 httplib2                      0.10.3     py27_0                      conda-forge/linux-64
 perl-gd                       2.56       0                           bioconda/linux-64   
 r-amap                        0.8_14     r3.2.2_0                    bioconda/linux-64   
 perl-posix                    1.38_03    pl5.22.0_0                  bioconda/linux-64   
 perl-math-cdf                 0.1        1                           bioconda/linux-64   
 weblogo                       2.8.2      0                           bioconda/linux-64   
 perl-xml-libxml               2.0124     0                           bioconda/linux-64   
 r-dendextend                  1.1.2      r3.2.2_0                    bioconda/linux-64   
 pygraphviz                    1.3.1      py27_0                      bioconda/linux-64   
 r-zoo                         1.7_14     r3.3.2_0                    conda-forge/linux-64
 perl-io-all                   0.86       0                           bioconda/linux-64   
 r-dynamictreecut              1.63_1     r3.3.1_0                    bioconda/linux-64   
 r-rjsonio                     1.3_0      r3.3.2_0                    conda-forge/linux-64
 r-jpeg                        0.1.8      r3.2.2_0                    bioconda/linux-64   
 seqlogo                       0.1.12     py35_0                      bioconda/linux-64   
 perl-readonly                 1.04       pl5.22.0_0                  bioconda/linux-64   
 perl-xml-dom                  1.45       0                           bioconda/linux-64   
 perl-soap-lite                1.19       0                           bioconda/linux-64   
 perl-algorithm-cluster        1.52       pl5.22.0_0                  bioconda/linux-64   
 bioconductor-ctc              1.52.0     r3.4.1_0                    bioconda/linux-64   
 perl-number-format            1.75       1                           bioconda/linux-64   
 perl-ole-storage_lite         0.19       0                           bioconda/linux-64   
 perl-digest-md5-file          0.08       pl5.22.0_0                  bioconda/linux-64   
 perl-mce-shared               1.836      pl5.22.0_0                  bioconda/linux-64   
 perl-statistics-distributions 1.02       0                           bioconda/linux-64   
 perl-object-insideout         4.04       pl526_0                     bioconda/linux-64   
 perl-spreadsheet-writeexcel   2.40       0                           bioconda/linux-64   
 vmatch                        2.3.0      h516909a_0                  bioconda/linux-64   
 perl-class-std-fast           0.0.8      0                           bioconda/linux-64   
 d3                            4.10.2     0                           conda-forge/noarch  
 perl-bio-das                  1.17       0                           bioconda/noarch     
 perl-email-simple             2.216      0                           bioconda/noarch     
 perl-ensembl-api              98         0                           bioconda/noarch     
 perl-ensembl-genomes          44         0                           bioconda/noarch     
 perl-http-tiny                0.076      pl526_0                     bioconda/noarch     
 perl-lockfile-simple          0.208      0                           bioconda/noarch     
 perl-rest-client              273        0                           bioconda/noarch     
 perl-util-properties          0.18       0                           bioconda/noarch     
 r-egg                         0.2.0      r3.3.2_0                    conda-forge/linux-64
 r-flux                        0.3_0      r3.3.2_0                    conda-forge/linux-64
 libgcc-ng                     9.3.0      h5dbcf3e_17                 conda-forge/linux-64
 libstdcxx-ng                  9.3.0      hdf63c60_16                 conda-forge/linux-64
 python                        3.6.9      h9d8adfe_0_cpython          conda-forge/linux-64
 perl                          5.26.2     h516909a_1006               conda-forge/linux-64
 r-base                        3.6.3      he766273_3                  conda-forge/linux-64
 libgfortran-ng                7.5.0      hdf63c60_9                  conda-forge/linux-64
```
My python version in conda is 3.9.4 as requested by mamba when installing via conda's ` -c conda-forge` .

I managed to install this package in another system with another miniconda installation, two days ago, after realizing it required both -c bioconda and -c conda-forge. I do not understand what may have changed since then. I have python 3.7.10 in that other system's miniconda, but it upgrades to 3.9.4 every time I install Mamba.

Thanks in advance!

**Comment by wolfv** [comment]
My suggestion is to remove defaults, and create new environments instead of installing into the base environment. It's a good practice to keep the base environment clean.

**Comment by wolfv** [comment]
you can create a new environment with `mamba create -n rsat rat-core -c bioconda -c conda-forge`

**Comment by apposada** [comment]
Hi again,

I understand. For clarity: i was not trying this in the base environment. I was aiming to, precisely, having separate environments specific for each tool, and keeping the environment clean. But I started for making the environment, installing mamba in that environment, and install anything else using mamba. I did not know it was possible to create environments from mamba itself.

It managed to install properly and it is up and running. Thanks and sorry for the inconvenience.
